storage/interface.go

```diff
@@ -145,6 +145,11 @@ type SelectHints struct {
 	Grouping []string // List of label names used in aggregation.
 	By       bool     // Indicate whether it is without or by.
 	Range    int64    // Range vector selector range in milliseconds.
+
+	// TrimDisabled allows to disable trimming matching series chunks based on query Start and End time.
+	// When disabled the returned samples may be outside the queried time range but Select() performances
+	// may be improved.
+	TrimDisabled bool
```

Comment:
Nit: `TrimDisabled` -> `DisableChunkTrimming`

Reply:
I'm fine with that too. Just to be on the same page, it trims result also when we return samples instead of chunks. Under the hood, we always trim chunks tho, but not sure that's an implementation detail we want to surface here.

Reply:
Introducing `Chunk` in the name starts going into the implementation details while I think the hints should stay agnostic to implementation. So I would either see `TrimDisabled` or `DisableTrimming`. So I think `TrimDisabled` is fine.

Reply:
Fair enough. I think `DisableTrimming` indicates that it's a hint to disable trimming. `TrimDisabled` is more passive, as if it was already decided, but it's really up to implementation whether hint is used or not. Anyway, not important.

Reply:
No strong opinion. `DisableTrimming` should be fine.